packages/gafl-webapp-service/src/handlers/__tests__/multibuy-duplicate-handler.spec.js

```diff
@@ -1,152 +1,40 @@
-        length: '12M',
-        start: '2025-01-01',
-        price: '30'
-      }
-    ]
-
-    const result = await checkDuplicates(licences)
-    expect(result).toBeFalsy()
+  const getLicenceTwo = () => ({
+    licenceHolder: 'Angler 2',
+    type: 'Coarse 3 rod',
+    length: '12M',
+    start: '2023-01-21',
+    price: 50
   })
-
-  it('should return false if no duplicate type', async () => {
-    const licences = [
-      {
-        licenceHolder: 'name',
-        type: 'Salmon and sea trout',
-        length: '12M',
-        start: '2025-01-01',
-        price: '30'
-      },
-      {
-        licenceHolder: 'name',
-        type: 'Trout and coarse, 2 rod',
-        length: '12M',
-        start: '2025-01-01',
-        price: '30'
-      },
-      {
-        licenceHolder: 'name',
-        type: 'Trout and coarse, 3 rod',
-        length: '12M',
-        start: '2025-01-01',
-        price: '30'
-      }
-    ]
-
-    const result = await checkDuplicates(licences)
-    expect(result).toBeFalsy()
+  const getLicenceThree = () => ({
+    licenceHolder: 'Angler 3',
+    type: 'Coarse 2 rod',
+    length: '12M',
+    start: '2023-02-12',
+    price: 50
   })
-
-  it('should return false if no duplicate length', async () => {
-    const licences = [
-      {
-        licenceHolder: 'name',
-        type: 'Salmon and sea trout',
-        length: '12M',
-        start: '2025-01-01',
-        price: '30'
-      },
-      {
-        licenceHolder: 'name',
-        type: 'Salmon and sea trout',
-        length: '1D',
-        start: '2025-01-01',
-        price: '30'
-      },
-      {
-        licenceHolder: 'name',
-        type: 'Salmon and sea trout',
-        length: '3D',
-        start: '2025-01-01',
-        price: '30'
-      }
-    ]
-
-    const result = await checkDuplicates(licences)
-    expect(result).toBeFalsy()
+  it('identifies when array has duplicate licences (same licenceHolder, type, length and start)', async () => {
+    const licenses = [getLicenceOne(), getLicenceOne(), getLicenceTwo()]
+    const result = await hasDuplicates(licenses)
+    expect(result).toBeTruthy()
   })
 
-  it('should return false if no duplicate start', async () => {
-    const licences = [
-      {
-        licenceHolder: 'name',
-        type: 'Salmon and sea trout',
-        length: '12M',
-        start: '2024-01-01',
-        price: '30'
-      },
-      {
-        licenceHolder: 'name',
-        type: 'Salmon and sea trout',
-        length: '12M',
-        start: '2023-01-01',
-        price: '30'
-      },
-      {
-        licenceHolder: 'name',
-        type: 'Salmon and sea trout',
-        length: '12M',
-        start: '2025-01-01',
-        price: '30'
-      }
-    ]
-
-    const result = await checkDuplicates(licences)
+  it('identifies when array has no duplicate licences', async () => {
+    const licenses = [getLicenceOne(), getLicenceTwo(), getLicenceThree()]
+    const result = await hasDuplicates(licenses)
```

Comment:
I think you've misunderstood what I meant when I said 'there should be two tests'. I mentioned using it.each, because you were writing the same test several times. So two tests, one to check arrays with duplicates and one to check arrays without duplicates. Then several different _cases_ passed to each test - I gave three licence holders as an example, but I'm not sure that's enough. What do you think? Do you feel it's well tested as is?

Reply:
I think when duplicate one test would be enough, but I will add multiple tests for when not duplicate based on if one area not matching (licence holder type etc)